%YAML:1.0

#--------------------------------------------------------------------------------------------
# Camera Parameters. Adjust them!
#--------------------------------------------------------------------------------------------

# Camera calibration and distortion parameters (OpenCV) 
Camera.fx: 554.254691191187
Camera.fy: 554.254691191187
Camera.cx: 320.5
Camera.cy: 240.5

Camera.k1: 0.0
Camera.k2: 0.0
Camera.p1: 0.0
Camera.p2: 0.0

Camera.width: 640
Camera.height: 480
Camera.max_dis: 5.0
Camera.min_dis: 0.5

# Camera frames per second 
Camera.fps: 20.0

# IR projector baseline times fx (aprox.)
Camera.bf: 40.0

# Color order of the images (0: BGR, 1: RGB. It is ignored if images are grayscale)
Camera.RGB: 0

# Close/Far threshold. Baseline times.
ThDepth: 40.0

# Deptmap values factor
DepthMapFactor: 1.0

# 1: robot_camera tf; 2: use imu     imu pose,qw:0.640856,qx:0.298836,qy:0.298836,qz:0.640856
# imu pose,qw:0.640856,qx:-0.298836,qy:-0.298836,qz:-0.640856
# qw:0.627211,qx:0.326505,qy:0.326505,qz:0.627211
#ConstraintType: 3
Tworld_camera.qw: -0.364187
Tworld_camera.qx: 0.606109
Tworld_camera.qy: 0.606109
Tworld_camera.qz: -0.364187

Tworld_camera.tx: 0.0
Tworld_camera.ty: 0.0
Tworld_camera.tz: 1.4  #this height is 0.2 higher than the height in gazebo

#translation from baselink frame to camera frame
#tum
#Trobot_camera.qx: 0.0003
#Trobot_camera.qy: 0.8617
#Trobot_camera.qz: -0.5072
#Trobot_camera.qw: -0.0145

#v1  -0.364187,qx:0.606109,qy:-0.606109,qz:0.364187
#gazebo camera pose,qw:-0.364187,qx:0.606109,qy:0.606109,qz:-0.364187
Trobot_camera.qw: -0.364187
Trobot_camera.qx: 0.606109
Trobot_camera.qy: -0.606109
Trobot_camera.qz: 0.364187

#kinectdk 20 5?
#Trobot_camera.qx: -0.506
#Trobot_camera.qy: 0.507
#Trobot_camera.qz: -0.493
#Trobot_camera.qw: 0.494

Trobot_camera.tx: 0.0
Trobot_camera.ty: 0.0
Trobot_camera.tz: 1.4  #this height is 0.2 higher than the height in gazebo
Trobot_camera.down_nbv_height: 0.0


#--------------------------------------------------------------------------------------------
# ORB Parameters
#--------------------------------------------------------------------------------------------

# ORB Extractor: Number of features per image0
ORBextractor.nFeatures: 1000

# ORB Extractor: Scale factor between levels in the scale pyramid 	
ORBextractor.scaleFactor: 1.2

# ORB Extractor: Number of levels in the scale pyramid	
ORBextractor.nLevels: 8

# ORB Extractor: Fast threshold
# Image is divided in a grid. At each cell FAST are extracted imposing a minimum response.
# Firstly we impose iniThFAST. If no corners are detected we impose a lower value minThFAST
# You can lower these values if your images have low contrast			
ORBextractor.iniThFAST: 20
ORBextractor.minThFAST: 7

#--------------------------------------------------------------------------------------------
# Viewer Parameters
#--------------------------------------------------------------------------------------------
Viewer.KeyFrameSize: 0.05
Viewer.KeyFrameLineWidth: 1
Viewer.GraphLineWidth: 0.9
Viewer.PointSize: 2
Viewer.CameraSize: 0.08
Viewer.CameraLineWidth: 3
Viewer.ViewpointX: 0
Viewer.ViewpointY: -0.7
Viewer.ViewpointZ: -1.8
Viewer.ViewpointF: 500

Viewer.pangolin: 0
Viewer.rviz: 1
Viewer.savelocalobject: 1

Viewer.show_object3d_frame: 0

Viewer.object_Duration: 0.1
Viewer.direction_Duration: 0.1
Viewer.IE_Duration: 0.1
Viewer.readlocalobject: 0
#--------------------------------------------------------------------------------------------
# Plane Extraction Parameters
#--------------------------------------------------------------------------------------------
Cloud.Dis: 3
Plane.MinSize: 500
Plane.AngleThreshold: 3.0
Plane.DistanceThreshold: 0.05
Plane.LeafSize: 0.01
Plane.MaxDistance: 10.0

#--------------------------------------------------------------------------------------------
# Plane Optimization Parameters
#--------------------------------------------------------------------------------------------
Plane.AssociationDisRef: 0.2
Plane.AssociationDisMon: 0.2
Plane.AssociationAngRef: 0.8 # 30 degree
Plane.AssociationAngMon: 0.8 # 30 degree
Plane.VerticalThreshold: 0.08716 # 85 degree
Plane.ParallelThreshold: 0.9962 # 5 degree
Plane.AngleInfo: 1
Plane.DistanceInfo: 100
Plane.Chi: 300
Plane.VPChi: 300
Plane.ParallelInfo: 0.5
Plane.VerticalInfo: 0.5


#--------------------------------------------------------------------------------------------
# Supposed Plane Extraction Parameters
#--------------------------------------------------------------------------------------------
Line.Ratio: 0.2
Line.DistanceThreshold: 0.01
Plane.ShowSPlane: 1

#--------------------------------------------------------------------------------------------
# Result File Paths
#--------------------------------------------------------------------------------------------
SavePath.Frame: "FrameTrajectory.txt"
SavePath.Keyframe: "KeyFrameTrajectory.txt"
SavePath.PlaneFeature: "PlaneFeature.txt"

#--------------------------------------------------------------------------------------------
# foreground object
#--------------------------------------------------------------------------------------------
IForest.Threshold: 0.6

#--------------------------------------------------------------------------------------------
# Local NBV
#--------------------------------------------------------------------------------------------
MAM.isused: 0
MAM.neckdelay: 100
MAM.divide: 20
MAM.NeckRange: 6.28
MAM.TfDuration: 2.0

#--------------------------------------------------------------------------------------------
# Global NBV
#--------------------------------------------------------------------------------------------

IE.rows: 18
IE.cols: 18
IE.P_occ: 0.6
IE.P_free: 0.4
IE.P_prior: 0.5
IE.ThresholdPointNum: 6
IE.PublishIEwheel: 0

#IE.BackgroudObjectNum: 2    #背景物体的先验数量
Plane.Height.Max: 2.0
Plane.Height.Min: 0.3
#Planle.Safe_radius: 0.3
Planle.Gobal_Candidate_Num: 36
NBV_Angle_correct_factor: 0 #-0.5
Candidate_num_topub: 1 #36
ViewField: 0.7



#核心控制选项
PubGlobalGoal: 0  # 为1，用于实体机器人，需调用movebase；为0时，用于gazebo环境，手动键盘控制。
PubLocalGoal: 1  # 1 my_nbv ; 2 cover;  0 frontier

ObserveMaxNumBackgroudObject: 100
MAM.Reward_angle_cost: 0.0  #计算reward时的dis。
MAM.Reward_dis: 0.5  #计算reward时的dis。当前没用到
Planle.Safe_radius: 0.5

ConstraintType: 1 #3 rostf; 2 imu; 1 yaml
FakeBackgroudObjects: 0  #fake背景物体的数量
IE.BackgroudObjectNum: 5   #背景物体的先验数量
IE.ThresholdEndMapping: 1.45   #0.4 #0.42   #设置为极小值，即可关闭end_build
#--------------------------------------------------------------------------------------------
# yolo
#         "person人类",  //0
#        "`b`icycle自行车", "car汽车", "motorcycle", "airplane", "bus",   //1
#        "train", "truck", "boat", "traffic light",   "fire hydrant", //6
#        "stop sign停止标", "parking meter", "bench", "bird", "cat", //11
#        "dog", "horse", "sheep", "cow",  "elephant", //16
#        "bear", "zebra", "giraffe", "backpack背包", "umbrella雨伞", //21
#        "handbag手提包", "tie领带", "suitcase手提箱", "frisbee",  "skis", //26
#        "snowboard", "sports ball", "kite", "baseball bat", "baseball glove", //31
#        "skateboard", "surfboard",  "tennis racket", "bottle瓶子", "wine glass酒杯", //36
#        "cup杯子", "fork", "knife", "spoon", "bowl碗", //41
#        "banana香蕉", "apple苹果",   "sandwich三明治", "orange橙子", "broccoli", //46
#        "carrot", "hot dog热狗", "pizza", "donut", "cake蛋糕", //51
#        "chair椅子", "couch沙发",  "potted plant盆栽", "bed床", "dining table餐桌",//56
#        "toilet", "tv电视", "laptop笔记本电脑", "mouse鼠标", "remote遥控器", //61
#        "keyboard键盘", "cell phone手机",  "microwave微波炉", "oven烤箱", "toaster烤面包机", //66
#        "sink水槽", "refrigerator冰箱", "book书", "clock钟", "vase花瓶", //71
#        "scissors", "teddy bear泰迪熊",  "hair drier", "toothbrush"};//76
#--------------------------------------------------------------------------------------------
#yolo_class: [24, 28, 39, 56, 57, 58, 59, 60, 62, 63, 66, 67, 73, 72, 11]
#yolo_class: [24, 28, 39, 56, 57, 58, 59, 60, 62, 63, 66, 67, 73, 72, 11]
#yolo_class: [77]
#77,  #泰迪熊
#73,  #书
#56,  #椅子
#39,  #瓶子
#63,  #笔记本电脑
#75,  #花瓶
#66,  #键盘
#41   #杯子
yolo_class: [77, 73,  39,  63, 75,  66,  41, 64]   #先去掉笔记本电脑 63
#yolo_class: [77, 73, 56, 39,  75,  66,  41]
